### DISCUSSION TITLE
MAVROS timeout when using BlueROV2 Heavy model

### DISCUSSION THREAD

**Discussion by evan-palmer** [Robotic-Decision-Making-Lab/blue#191] [Q&A]
_Originally posted [here](https://github.com/Robotic-Decision-Making-Lab/blue/discussions/45#discussioncomment-9875184) by @SusannaMunari_

Hello,

I am encountering the same issue when running this:
ros2 launch blue_bringup bluerov2_heavy.launch.yaml use_sim:=true

I get this output, and I am not able to connect and simply echo the topics.

![342895757-41f49262-6154-42fd-b601-519ed4af6a43](https://github.com/Robotic-Decision-Making-Lab/blue/assets/75103118/1716df57-ee9a-4e3c-91d4-f2557cedf49c)

Curiously, the same process works smoothly when I try with the normal configuration:
ros2 launch blue_bringup bluerov2.launch.yaml use_sim:=true

With this I am connected, I can access the topics and change flight mode.

Is there any configuration that I am missing?

**Comment by evan-palmer** [comment]
It appears as though you found a bug in the launch file :smiley: I'll make a new issue and get a fix pushed for that. Are you using Humble or Iron? I can prioritize the one that you are using

**Reply by SusannaMunari** [reply]
Thank you for the quick answer!
I am running Iron.

As a side node, I had the exact same issue when I created a simple workspace to access topics directly from the BlueROV2 Heavy that I am working on. I tried using Blue for this reason. It might be just a coincidence, but it is odd.

**Reply by evan-palmer** [reply]
When using MAVROS on hardware, I've found that it occasionally has issues connecting, which results in the issue that you shared. In some cases it can be a bad network configuration, other times it's because of high latency in the connection (if you are connecting via tether), and sometimes it just flat-out fails! If it's persistent, then I would be inclined to believe that it's a bad network configuration or latency over tether (if you are connecting via tether)

**Comment by evan-palmer** [comment]
This should be fixed now! Let me know if you run into any other issues!

**Reply by SusannaMunari** [reply]
Thank you!
Sorry for the late response, I could only get back to it now. 
I am installing from source now with the Dockerfile, as suggested [here](https://robotic-decision-making-lab.github.io/blue/installation/), but I am getting some errors when building the container that are most likely due to some path error.

After cloning the repo, I just run this command from the .docker folder:
`docker build -t my_blue_heavy .`

For example, I  had issues with this line of the Dockerfile:
```
# Install the Python requirements that aren't available as rosdeps
RUN python3 -m pip install -r $(pwd)/src/blue/requirements-build.txt
```
I changed it to:
`RUN python3 -m pip install qtm`
because I couldn't find the right path to the file. 

Now I see the same issue:
2.732 /bin/sh: 1: cannot open src/blue/blue.repos: No such file

Am I missing something on the directory structure?

**Reply by evan-palmer** [reply]
When building the Docker image, you would run something like this from the `blue/` directory:
```
docker build -f .docker/Dockerfile -t my_blue_heavy .
```
Though, if you want to use Docker, I would recommend using the provided devcontainer (easiest and recommended) or the docker compose script. If you would like to install from source (locally, then I recommend using the Dockerfile as a reference

**Reply by SusannaMunari** [reply]
Thank you again! And great work :+1:
